packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -3991,7 +4079,11 @@ if (__DEV__) {
     ): T {
       currentHookNameInDev = 'useSyncExternalStore';
       updateHookTypesDev();
-      return updateSyncExternalStore(subscribe, getSnapshot, getServerSnapshot);
+      return rerenderSyncExternalStore(
```

Comment:
(also noticed that these should also maybe wrap in InvalidNestedHooksDispatcher although idk why you would try to use a hook in getSnapshot)